samtranslator/plugins/api/implicit_rest_api_plugin.py

```diff
@@ -87,10 +89,18 @@ def _process_api_events(self, function, api_events, template, condition=None):
             if isinstance(api_id, dict):
                 raise InvalidEventException(logicalId, "Api Event must reference an Api in the same template.")
 
-            api_dict = self.api_conditions.setdefault(api_id, {})
-            method_conditions = api_dict.setdefault(path, {})
+            api_dict_condition = self.api_conditions.setdefault(api_id, {})
+            method_conditions = api_dict_condition.setdefault(path, {})
             method_conditions[method] = condition
 
+            api_dict_deletion = self.api_deletion_policies.setdefault(api_id, {})
+            method_deletion_policies = api_dict_deletion.setdefault(path, {})
+            method_deletion_policies[method] = deletion_policy
+
+            api_dict_update_replace = self.api_update_replace_policies.setdefault(api_id, {})
+            method_update_replace_policies = api_dict_update_replace.setdefault(path, {})
+            method_update_replace_policies[method] = update_replace_policy
```

Comment:
Same as above, not sure if we need method level `DeletionPolicy` and `UpdateReplacePolicy` values